src/protean/core/field/association.py

```diff
@@ -109,5 +154,110 @@ def _cast_to_type(self, value):
             self.fail('invalid', value=value)
         return value
 
+
+class Association(FieldCacheMixin, Field):
```

Comment:
Association need not be a sub class of Field...

Reply:
You are right. Good thought. I moved the common descriptor functionality I needed into a mixin as part of this commit: https://github.com/proteanhq/protean/pull/86/commits/f795469d53adfe00e8bf5167d0e38a0e933b8547